### PR TITLE
Week That Was (mechanical stages) + weekly audio cadence for existing shows (skill v3.0)

### PR DESCRIPTION
**Pull request by dhk**
## Summary

Two related but independent pieces of work on top of the reading-list-builder pipeline:

**1. "Week That Was" — mechanical stages**
- `reading-with-ears/scripts/week_that_was.py`: pre-flight (locates the ISO week's daily run YAMLs, flags missing days, fails fast if `ANTHROPIC_API_KEY` is unset), deterministic zeitgeist tag counting with a noise-floor trend classification (emerging/growing/dipping/steady/low-volume), Sonnet section synthesis with per-section resume granularity in the manifest, and Markdown doc assembly. Ships without the zeitgeist narrative or article-ideas sections (those need a writer profile that doesn't exist yet) — see `docs/week-that-was-design.md` for the full design and build order.
- `bin/rwe-weekly`: PID-lock guarded wrapper, same shape as `rwe-publish`.
- `reading-with-ears/config/weekly.json`: 6-topic taxonomy + weekly show config (Element.fm show ID still TBD).
- Tested end-to-end with `--dry-run` against real reading-db data, including a resume-after-partial-failure case.

**2. Weekly audio cadence for the four existing shows (news/think/professional/vital-signs)**
- Daily gather/synthesize is unchanged, but audio generation/poll/title/download/publish no longer happen every day.
- `reading-list-builder` skill bumped to v3.0: each feed now accumulates a full ISO week of sources into one NotebookLM notebook (STEP 3), with a one-time reconciliation procedure for feeds mid-week under the old per-day scheme. A separate, explicitly-triggered `WEEKLY AUDIO FLOW` (STEPS W1-W4) generates and publishes one episode per feed per week to the same existing Element.fm shows.
- Light mode dropped (its only differentiator — same-day audio — no longer applies to any mode).
- `bin/rwe-run.sh` / `bin/rwe-catchup.sh` updated to match (no longer call `publish_episodes.py` themselves); new `bin/rwe-weekly-audio` wrapper, manual trigger only.
- Fixes a pre-existing syntax error in `rwe-catchup.sh`'s date-range loop (`<=` isn't valid in bash `[[ ]]` string comparison), found via `bash -n` while touching that file.
- Full design and rationale in `docs/weekly-cadence-migration.md`, including the reconciliation procedure.

This PR does **not** touch `feeds.json` show identity/IDs, and does not change `publish_episodes.py`/`podcast_config.py` — the existing per-date filename convention works unchanged for weekly episodes (see migration doc §4).

## Test plan
- [x] `week_that_was.py --dry-run` against real `dhkondata/reading-db/runs/` data — verified missing-day handling, tag classification, and manifest resume (including a fixed edge case where a resumed real run must still fail fast on a missing API key even if `preflight` was already marked complete by an earlier dry run)
- [x] `bash -n` syntax check on all new/modified shell scripts
- [ ] Live NotebookLM MCP run of STEP 3's weekly-notebook lookup and reconciliation path — not exercised in this environment (no MCP connection available here); recommend a manual dry run against this week's real notebooks before relying on it
- [ ] `week_that_was.py` section synthesis (Sonnet calls) — not exercised without `ANTHROPIC_API_KEY`/`anthropic` package in this environment; dry-run path is verified, live path is not
- [ ] `bin/rwe-weekly-audio` end-to-end against a real week with existing notebooks

---
_Generated by [Claude Code](https://claude.ai/code/session_01QpyDSUB41GjR4yTyADpCvx)_

### CODE REVIEW

File: reading-with-ears/scripts/week_that_was.py [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+                    "thread_id": email.get("thread_id"),
+                    "url_canonical": (art.get("source") or {}).get("url_canonical"),
+                    "bullets": synthesis.get("bullets") or [],
+                    "confidence_note": synthesis.get("confidence_note"),
+                    "tags": art.get("tags") or [],
+                }
+
+
+def articles_for_section(all_articles, feeds):
+    feeds = set(feeds)
```
Comment (dhk) [inline reading-with-ears/scripts/week_that_was.py:189]: **Every section will always be empty — `label` and `feeds` are different vocabularies.**

`a["label"]` (line 113/604) comes straight from the daily YAML's `emails[].label` field, which per SKILL.md STEP 7's own example is the raw Gmail label string, e.g. `"newsletter/pro"` (not the feed slug). But `weekly.json`'s taxonomy `feeds` lists use feed **slugs** — `"professional"`, `"think"`, `"news"`, `"vital-signs"` (matching `feeds.json`'s `slug` field).

```python
def articles_for_section(all_articles, feeds):
    feeds = set(feeds)
    return [a for a in all_articles if a["label"] in feeds]
```

`"newsletter/pro" in {"professional", "think", "news"}` is always `False`. Since none of the raw Gmail labels (`newsletter/news`, `newsletter/think`, `newsletter/pro`, `newsletter/healthcare`) ever equal any feed slug, `articles_for_section` returns `[]` for every section, every week — every section takes the "no articles this week" skip path (line ~440-443) regardless of how much mail actually came in. The whole synthesis feature silently produces an empty doc (just the zeitgeist table) forever.

Fix: either map `a["label"]` through the feed's `gmail_labels` → `slug` before comparing, or use `article.routing.category`/notebook slug instead of the raw email label.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: reading-with-ears/skills/user/reading-list-builder/SKILL.md [hunk 2]
```diff
@@ -106,34 +111,55 @@ Note in the report.
...
+   reconciliation before proceeding — see "Reconciliation" below. After
+   reconciliation there is exactly one canonical notebook_id for this feed this
+   week; use it.
+
+### Notebook naming
 
-Notebook naming convention:
 ```
-reading-list-YYYY-MM-DD-nn <notebook_category>
+reading-list-YYYY-Www-nn <notebook_category>
```
Comment (dhk) [inline reading-with-ears/skills/user/reading-list-builder/SKILL.md:157]: **This breaks `rwe-publish`'s notebook lookup — the weekly audio flow can never find its notebooks.**

`reading-with-ears/scripts/podcast_config.py` (unchanged by this PR) hardcodes:

```python
READING_LIST_NOTEBOOK_RE = re.compile(
    r"^reading-list-(?P<date>\d{4}-\d{2}-\d{2})-(?P<nn>\d{2})\s+(?P<category>.+)$"
)
```

and `publish_episodes.py`'s `find_notebooks_for_date()` filters candidates with `title.startswith(f"reading-list-{target_date}")` before even running that regex, where `target_date` is a `YYYY-MM-DD` string.

With the new naming convention here (`reading-list-YYYY-Www-nn ...`, e.g. `reading-list-2026-W26-01 📰 News & Current Affairs`), no notebook title will ever start with `reading-list-2026-07-03` or match `\d{4}-\d{2}-\d{2}` — the week label doesn't fit that shape. So when STEP W3 hands off via `rwe-publish --date YYYY-MM-DD ...`, `wait_for_notebooks_for_date`/`find_notebooks_for_date` will find **zero** notebooks, `rwe-publish` will warn "No notebooks found before timeout/discovery step." and `sys.exit(1)` — the entire weekly audio flow fails at STEP W3 every single week.

The PR description says "`publish_episodes.py`/`podcast_config.py`: no changes needed" — that's true for the *downloaded audio filename* convention (`YYYY-MM-DD-slug.mp3`, untouched), but the *notebook title* matching regex is a separate mechanism that this PR's naming change breaks. `podcast_config.py`'s `READING_LIST_NOTEBOOK_RE`/`parse_reading_list_notebook_title` and `publish_episodes.py`'s `find_notebooks_for_date`/`cleanup_old_items` all need to understand the new `YYYY-Www` notebook title format.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: reading-with-ears/scripts/week_that_was.py [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+            continue
+
+        articles = articles_for_section(all_articles, feeds)
+        if not articles:
+            sec_state.update(status="skipped", reason="no articles this week")
+            save_manifest(manifest_path, manifest)
+            continue
+
+        if args.dry_run:
+            sec_state.update(status="skipped", reason="dry-run")
```
Comment (dhk) [inline reading-with-ears/scripts/week_that_was.py]: **A `--dry-run` permanently poisons a section as "skipped," blocking all future real synthesis for it.**

```python
if sec_state.get("status") == "skipped":
    continue          # line 447
...
if args.dry_run:
    sec_state.update(status="skipped", reason="dry-run")   # line 457
    save_manifest(manifest_path, manifest)
    continue
```

The resume check at line 447 treats `status == "skipped"` as terminal regardless of *why* it was skipped — it doesn't distinguish `reason="dry-run"` from `reason="no articles this week"`. So: run `week_that_was.py --dry-run` once (exactly what the PR's own test plan did — "`week_that_was.py --dry-run` against real `dhkondata/reading-db/runs/` data"), and every section with articles gets `status="skipped", reason="dry-run"` written to `manifest.yaml`. Run the *real* pipeline afterward for the same week, and every one of those sections hits the line-447 `continue` before ever reaching `synthesize_section` — no Sonnet call happens, `section_texts` stays empty for that section, and the stage still reports `status: complete` (since `"skipped"` counts as done in the `all(s in ("complete", "skipped") ...)` check a few lines down). The final doc silently ships with no section content and no error.

This is the same class of bug the PR explicitly patched for `preflight` (the comment right above `if not args.dry_run and not os.environ.get(...)`: "a prior `--dry-run` can leave preflight marked complete... a real run must still fail fast rather than inherit that") — but the fix wasn't applied to `section_synthesis`. Suggest giving dry-run skips a distinct status (e.g. `"dry_run_skipped"`) that's excluded from the terminal-skip check, or clearing dry-run-skip entries at the start of a non-dry-run invocation.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: reading-with-ears/scripts/week_that_was.py [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+import sys
+import time
+from datetime import date, datetime, timezone
+from pathlib import Path
+
+import yaml
+
+# ── Paths ──────────────────────────────────────────────────────────────────
+
+SCRIPT_DIR = Path(__file__).resolve().parent
```
Comment (dhk) [inline reading-with-ears/scripts/week_that_was.py]: **Repo-root discovery here doesn't match the rest of the codebase, and breaks under `SYNC_MODE=copy`.**

```python
SCRIPT_DIR = Path(__file__).resolve().parent
RWE_ROOT = SCRIPT_DIR.parent
REPO_ROOT = RWE_ROOT.parent
RUNS_DIR = REPO_ROOT / "dhkondata" / "reading-db" / "runs"
```

Every other entry point in this repo resolves the repo root via `bin/rwe-common.sh`'s `rwe_repo_root()` (which honors `RWE_REPO`, then `~/.config/reading-with-ears/config.json`'s `repo_root`, then falls back to a relative-to-caller guess) — precisely because `install-local.sh` can sync scripts to `~/.local/share/reading-with-ears/scripts/` either as symlinks (default) or as real copies (`--copy` / `sync_mode: "copy"` in config.json, both explicitly supported).

Under the default symlink mode, `Path(__file__).resolve()` follows the symlink back to the real repo path, so this happens to work. But under copy mode — a first-class documented option in `install-local.sh` — `week_that_was.py` becomes a real file at `~/.local/share/reading-with-ears/scripts/week_that_was.py`, `.resolve()` has nothing to follow, and `REPO_ROOT` ends up as `~/.local/share`, so `RUNS_DIR` points at `~/.local/share/dhkondata/reading-db/runs` — a path that doesn't exist. The script would immediately fail with "no daily runs found for {week_str}" even though real run data exists in the actual repo checkout. `bin/rwe-weekly` (which invokes this script) has no equivalent of `rwe_repo_root()`/`RWE_REPO` to paper over this either.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by dhk — COMMENTED** [review]
Reviewed the two pieces (Week That Was mechanical stages + weekly audio cadence migration). The design docs and mechanical scaffolding (manifest resume, noise-floor trend classification, PID-locked wrapper scripts) are solid, but I found two cross-file breaks that make both new flows non-functional as written, plus a manifest-state bug and a path-resolution fragility. Left 4 inline comments, most severe first:

1. **`week_that_was.py:130`** — `articles_for_section` filters on `a["label"]` (raw Gmail label, e.g. `"newsletter/pro"`) against `weekly.json`'s `feeds` lists (feed slugs, e.g. `"professional"`). These never intersect, so every section is always "skipped — no articles this week," every week — the whole synthesis feature silently produces an empty doc.
2. **`SKILL.md:144`** — the new weekly notebook-title convention (`reading-list-YYYY-Www-nn ...`) doesn't match `podcast_config.py`'s `READING_LIST_NOTEBOOK_RE` (still `YYYY-MM-DD`), which `publish_episodes.py`'s `find_notebooks_for_date()` depends on. The weekly audio flow's STEP W3 hand-off to `rwe-publish` will find zero notebooks and exit 1, every week.
3. **`week_that_was.py:457`** — a `--dry-run` (which the PR's own test plan ran) permanently marks sections `status="skipped"`, and the resume check at line 447 treats any `"skipped"` status as terminal — so a later real run never actually synthesizes those sections, and the stage still reports `complete`. Same class of bug the PR already fixed for `preflight`, just not applied to `section_synthesis`.
4. **`week_that_was.py:35`** — repo-root discovery via `Path(__file__).resolve()` bypasses the `rwe_repo_root()`/`RWE_REPO` mechanism every other entry point uses, and breaks under the documented `SYNC_MODE=copy` install option.

Given #1 and #2 each independently break one of the PR's two headline features end-to-end, I'd suggest resolving those before merge.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by dhk** [comment]
All 4 review findings confirmed and fixed in 85f16cf:

1. **Label/slug mismatch** — added `build_label_to_slug()` (from `feeds.json`'s `slug` + `gmail_labels`) and normalize every article's label through it in `iter_articles`. Verified against real data: week 2026-W18 (old `newsletter/xxx`-format labels) now correctly attributes all 11 articles to their feed slugs — previously zero.
2. **Notebook regex break** — added `READING_LIST_WEEKLY_NOTEBOOK_RE`/`parse_reading_list_weekly_notebook_title` to `podcast_config.py`, `find_notebooks_for_week`/`wait_for_notebooks_for_week` to `publish_episodes.py`, and a `--notebook-week` flag (separate from `--date`, which still controls the downloaded filename). STEP W3 and `rwe-weekly-audio` updated to pass it.
3. **Dry-run poisoning** — dry-run skips now get a distinct `dry_run_skipped` status that's only terminal for another dry run, not a real one. Verified: dry-run → dry-run → real (no key) now correctly re-attempts rather than silently reporting the stage complete with no content.
4. **Repo-root fragility** — `week_that_was.py` now prefers `RWE_REPO` if set; `bin/rwe-weekly` exports it after resolving via `rwe_repo_root()`, which already handles the copy-mode case.

Also found one more while verifying fix #1 against real data (not part of the original review): `2026-05-02.yaml` (written by `reading-db-backfill.py`, `pipeline_mode: deep-email-body`) stores `synthesis` as a flat list of strings rather than the current `{bullets, confidence_note}` dict — crashed `iter_articles` with an `AttributeError`. Added a small shape-handling helper for both forms.

Re-ran the full regression after fixes: `py_compile`/`bash -n` clean on all touched files, dry-run → dry-run → real-without-key behaves correctly, and the label fix is confirmed against real historical data rather than just the one file that happened to already use the slug form.


---
_Generated by [Claude Code](https://claude.ai/code)_
